### PR TITLE
earn group/online/offline members sign up

### PR DESCRIPTION
**Pull request by RamyAlshurafa**
#### Issues this PR relates to
Relates #48 , #88 (need to be linked with front end)
---

#### Any info the reviewer needs to know to test 
e.g. Add env vars, run npm install

---

#### Any info the team needs to consider before merged into development
e.g. Will a database need to be rebuilt (and is this ok if existing data), add env vars

---

#### Quick Pull Request Checklist
Please make sure you ensure the following has been done before marking your PR as awaiting-review. For full checklist and process review, please refer to Process Guide (#8)
- Branch is up to date with develop
- Project runs
- You can download and install locally, and run the project locally
- All existing and new features run without bugs
- Checked that features/design works across all agreed browsers + devices
- The new work has been checked with Axe accessibility checker and doesn't throw significant errors  
- Linter is used
- Code is consistent with agreed team style
- Naming of files is consistent with project 
- Tests all pass
- No console logs

### CODE REVIEW

File: client/src/validation/err-msgs.js [hunk 1]
```diff
@@ -4,3 +4,4 @@ export const SHORT_PASSWORD =
 export const INVALID_EMAIL = "invalid email";
 export const INVALID_ROLE = "invalid role";
 export const NO_PARTICIPANTS = "You should select participants number";
+export const SHOULD_AGREE_ON_TERMS = "You have to agree on this"; // TODO find good expression :)
```
Comment (dupreesi) [inline client/src/validation/err-msgs.js:7]: You must agree to our terms and conditions :)

File: server/services/errorHandler/messages.js [hunk 2]
```diff
@@ -81,6 +81,12 @@ const errors = {
       httpStatusCode: 404,
       data,
     }),
+  EMAIL_ALREADY_USED: (data) =>
+    new CError({
+      message: 'this email already used',
```
Comment (dupreesi) [inline server/services/errorHandler/messages.js:86]: this email already exists

File: client/src/api-calls/auth.js [hunk 3]
```diff
@@ -29,7 +29,15 @@ const tempraryLogin = async ({ id, role, options }) => {
...
+    const { data } = await axios.post(`${AUTH_BASE}/signup`, formData);
+    return { data };
+  } catch (error) {
+    const err = handleError(error, options);
+    return { error: err };
+  }
+};
 const logout = async () => {};
 
 export { login, signup, logout, tempraryLogin };
```
[the remark this answers is not in the document]
Comment (RamyAlshurafa) [reply client/src/api-calls/auth.js:43]: done

File: client/src/pages/Signup/actions.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        payload: error,
+      });
+    }
+  } catch (error) {
+    dispatch({
+      type: GET_EARN_GROUP_INFO_BY_TOKEN_FAIL,
+      payload: error,
+    });
+  }
+};
```
[the remark this answers is not in the document]
Comment (RamyAlshurafa) [reply client/src/pages/Signup/actions.js:32]: done

File: client/src/validation/err-msgs.js (hunk 1 above)
Comment (RamyAlshurafa) [reply client/src/validation/err-msgs.js:7]: done

File: server/services/errorHandler/messages.js (hunk 2 above)
Comment (RamyAlshurafa) [reply server/services/errorHandler/messages.js:86]: done

**Pull request merged**